crates/scanner/src/cache.rs

```diff
@@ -67,7 +71,18 @@ impl TagCache {
 
                                     Ok(CacheLoadResult::Rebuild)
                                 } else {
-                                    Ok(CacheLoadResult::Loaded(cache))
+                                    // Check if wordlist hash matches
+                                    let current_wordlist_hash = compute_wordlist_hash();
+                                    if cache.wordlist_hash != current_wordlist_hash {
+                                        info!(
+                                            "Cache is out of date due to wordlist changes, rebuilding (cache hash: {:08x}, current hash: {:08x})",
+                                            cache.wordlist_hash,
+                                            current_wordlist_hash
+                                        );
+                                        Ok(CacheLoadResult::WordlistChanged)
+                                    } else {
+                                        Ok(CacheLoadResult::Loaded(cache))
+                                    }
```

Comment:
The cache shouldn't be rebuilt automatically when the wordlist changes, as it is a long procedure. When the game files change, a rebuild is always triggered automatically since the cache *must* be up to date with the game files.
We should only show an informative text to the user instructing them to rebuild the cache at their convenience